src/client/api/mgmt.c

```diff
@@ -278,14 +278,25 @@ daos_pool_remove_replicas(const uuid_t uuid, const char *group,
 }
 
 int
-daos_mgmt_list_pools(const char *group, daos_mgmt_pool_info_t *pools,
-		     daos_size_t *npools, daos_event_t *ev)
+daos_mgmt_list_pools(const char *group, daos_size_t *npools,
+		     daos_mgmt_pool_info_t *pools, daos_event_t *ev)
 {
 	daos_mgmt_list_pools_t	*args;
 	tse_task_t		*task;
 	int			 rc;
 
 	DAOS_API_ARG_ASSERT(*args, MGMT_LIST_POOLS);
+
+	if (npools == NULL) {
+		D_ERROR("npools must be non-NULL\n");
+		return -DER_INVAL;
+	}
+
+	if ((*npools == 0) && pools) {
+		D_ERROR("npools=0 but non-NULL pools=%p\n", pools);
+		return -DER_INVAL;
+	}
```

Comment:
I think this case is okay and could be handled consistently as "buffer too small" cases? *npools > 0 && pools == NULL might be worth an error.

Reply:
Removed the error check. Client provides a buffer but tells the server it has 0 entries available to fill in a response. API in this case can return rc=0 and provide the number of pools while not filling any of the client buffer positions.

The other case with *npools > 0 and pools == NULL I think we will handle as a (legal/successful) scenario where the caller (by providing pools=NULL) is asking for only the number of pools to be returned. The input value *npools > 0 may occur due to an uninitialized variable (for example). So we want this to succeed. Updated the implementation to pass npools=0 in the RPC request.